prow/spyglass/gcsartifact.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spyglass
+
+import (
+	"compress/gzip"
+	"context"
+	"io/ioutil"
+
+	"cloud.google.com/go/storage"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/spyglass/viewers"
+)
+
+// GCSArtifact represents some output of a prow job stored in GCS
+type GCSArtifact struct {
+	// The handle of the object in GCS
+	handle *storage.ObjectHandle
+
+	// The direct link to the Artifact, can be used for read operations
+	link string
+
+	// The path of the Artifact within the job
+	path string
+
+	// Max size to read before failing
+	sizeLimit int64
+}
+
+// NewGCSArtifact returns a new GCSArtifact with a given handle, canonical link, and path within the job
+func NewGCSArtifact(handle *storage.ObjectHandle, link string, path string) *GCSArtifact {
+	return &GCSArtifact{
+		handle:    handle,
+		link:      link,
+		path:      path,
+		sizeLimit: 500e6, // 500MB max file size
```

Comment:
comment on configuring this is outstanding

Reply:
also help admins determine a good value -- explain how to get to a good max size from their deployment strategy

Reply:
sgtm, I'm going to make this configurable

Reply:
size limit is now a configurable in spyglass (also explained how to choose a good limit)